components/needs_team/needs_team.jsx

```diff
@@ -81,35 +84,104 @@ export default class NeedsTeam extends React.Component {
             team,
             finishedFetchingChannels: false,
+            attemptingToJoinTeam: !team,
         };
+
+        if (!team) {
+            this.joinTeam(this.props);
+        }
     }
 
     UNSAFE_componentWillReceiveProps(nextProps) { // eslint-disable-line camelcase
         if (this.props.match.params.team !== nextProps.match.params.team) {
-            this.setState({team: this.updateCurrentTeam(nextProps)});
+            const team = this.updateCurrentTeam(nextProps);
+            this.setState({
+                team,
+                attemptingToJoinTeam: !team,
+            });
+            if (!team) {
+                this.joinTeam(nextProps);
+            }
         }
     }
 
-    updateCurrentTeam(props) {
-        // First check to make sure you're in the current team
-        // for the current url.
-        const teamName = props.match.params.team;
-        const team = TeamStore.getByName(teamName);
+    componentDidMount() {
+        startPeriodicStatusUpdates();
+        startPeriodicSync();
 
-        if (!team) {
-            props.history.push('/?redirect_to=' + encodeURIComponent(props.location.pathname));
-            return null;
+        // Set up tracking for whether the window is active
+        window.isActive = true;
+        Utils.applyTheme(this.props.theme);
+
+        if (UserAgent.isIosSafari()) {
+            // Use iNoBounce to prevent scrolling past the boundaries of the page
+            iNoBounce.enable();
         }
 
-        // If current team is set, then this is not first load
-        // The first load action pulls team unreads
-        if (TeamStore.getCurrentId()) {
-            this.props.actions.getMyTeamUnreads();
+        window.addEventListener('focus', this.handleFocus);
+        window.addEventListener('blur', this.handleBlur);
+        window.addEventListener('keydown', this.shortcutKeyDown);
+    }
+
+    componentDidUpdate(prevProps) {
+        const {theme} = this.props;
+        if (!Utils.areObjectsEqual(prevProps.theme, theme)) {
+            Utils.applyTheme(theme);
+        }
+    }
+
+    componentWillUnmount() {
+        stopPeriodicStatusUpdates();
+        stopPeriodicSync();
+        if (UserAgent.isIosSafari()) {
+            iNoBounce.disable();
         }
 
-        TeamStore.saveMyTeam(team);
-        BrowserStore.setGlobalItem('team', team.id);
-        TeamStore.emitChange();
+        clearInterval(wakeUpInterval);
+        window.removeEventListener('focus', this.handleFocus);
+        window.removeEventListener('blur', this.handleBlur);
+        window.removeEventListener('keydown', this.shortcutKeyDown);
+    }
+
+    handleBlur = () => {
+        window.isActive = false;
+        this.blurTime = new Date().getTime();
+        if (this.props.currentUser) {
+            this.props.actions.viewChannel('');
+        }
+    }
+
+    handleFocus = async () => {
+        await this.props.actions.markChannelAsRead(this.props.currentChannelId);
```

Comment:
It's not using the return `data` of [markChannelAsRead](https://github.com/mattermost/mattermost-redux/blob/master/src/actions/channels.js#L1101).  Will it work the same without await?

Reply:
True. This does not require await here